### PR TITLE
Geirst/fix bitvector search cache for imported attributes

### PR DESCRIPTION
**Pull request by geirst**
@toregge and @baldersheim please review

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/visibilityhandler.cpp [hunk 1]
```diff
@@ -50,23 +50,23 @@ void VisibilityHandler::commitAndWait()
     _writeService.summary().sync();
 }
 
-bool VisibilityHandler::startCommit(const LockGuard & unused)
+bool VisibilityHandler::startCommit(const LockGuard &unused, bool force)
 {
     (void) unused;
     SerialNum current = _serial.getSerialNum();
-    if (current > _lastCommitSerialNum) {
+    if (current > _lastCommitSerialNum || force) {
```
Comment (baldersheim) [inline searchcore/src/vespa/searchcore/proton/server/visibilityhandler.cpp]: I prefer parentheses so you do not need complete operator precedence ruleset control...

File: searchcore/src/vespa/searchcore/proton/server/visibilityhandler.cpp [hunk 2]
```diff
@@ -50,23 +50,23 @@ void VisibilityHandler::commitAndWait()
...
     return false;
 }
 
-void VisibilityHandler::performCommit()
+void VisibilityHandler::performCommit(bool force)
 {
     // Called by master thread
     SerialNum current = _serial.getSerialNum();
-    if (current > _lastCommitSerialNum) {
+    if (current > _lastCommitSerialNum || force) {
```
Comment (baldersheim) [inline searchcore/src/vespa/searchcore/proton/server/visibilityhandler.cpp]: Parentheses

File: searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp [hunk 3]
```diff
@@ -110,6 +110,7 @@ struct MyMetricsWireService : public DummyWireService
 struct MyDocumentDBReferenceResolver : public IDocumentDBReferenceResolver {
     std::unique_ptr<ImportedAttributesRepo> resolve(const search::IAttributeManager &,
 													const search::IAttributeManager &,
+													const std::shared_ptr<search::IDocumentMetaStoreContext> &,
```
Comment (baldersheim) [inline searchcore/src/tests/proton/documentdb/document_subdbs/document_subdbs_test.cpp]: Here is something odd with alignment, tabs versus spaces or just plain off.

**Review by baldersheim — APPROVED** [review]
Fine, to merge, just minor comments.

**Review by toregge — APPROVED** [review]
LGTM

**Pull request merged**